File: javabasic/reflect/src/com/sunny/model/Person.java
package com.sunny.model;

import java.io.InputStream;

public class Person {
	private int age;
	private String name;
	public static String sex;
	public int weight;

	public int getAge() {
		return age;
	}

	public Person() {
		super();
	}

	public Person(int age, String name) {
		super();
		this.age = age;
		this.name = name;
	}

	public Person(int age, String name, int weight,String sex) {
		super();
		this.age = age;
		this.name = name;
		this.weight = weight;
		this.sex = sex;
	}

	@SuppressWarnings("unused")
	private Person(int age) {
		super();
		this.age = age;
	}

	public void setAge(int age) {
		this.age = age;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public void showName(String name) {
		System.out.println(name);
	}

	public void showName() {
		System.out.println("alice");
	}

	@SuppressWarnings("rawtypes")
	public Class[] getClassNum(String name, int[] password) {
		return new Class[] { String.class };
	}

	@SuppressWarnings("unused")
	private void addInputStream(InputStream in) {
		System.out.println(in);
	}

	public static void eat(int num) {
		System.out.println("eat number:" + num);
	}

	@Override
	public String toString() {
		return "Person [age=" + age + ", name=" + name + "]";
	}
	
	public static void main(String[] args) {
		System.out.println("main");
	}
}

File: PreviousProjects/C++Project/继承/Person.cpp
#include<iostream>
#include<string.h>
using namespace std;
class Person
{
	public:
		int age;
		string name;
		Person()
		{
			cout<<"Person()"<<endl;
		}
		~Person()
		{
			cout<<"~Person()"<<endl;
		}
		void setAge(int newage)
		{
			age=newage;
		}
		void eat()
		{
			cout<<"eat()"<<endl;
		}
		int getAge()
		{
			return age;
		}
		void setName(string newname)
		{
			name=newname;
		}
		string getName()
		{
			return name;
		}
};
class Worker:public Person
{
	public:
		int salary;
		Worker()
		{
			cout<<"Worker()"<<endl;
		}
		~Worker()
		{
			cout<<"~Worker()"<<endl;
		}
};
int main(void)
{
	Worker *w=new Worker();
	w->setAge(22);
	w->setName("jin");
	w->eat();
	w->salary=1000;
	delete w;
	w=NULL;
	cout<<"______"<<endl;
	Worker w1;
	w1.setAge(22);
	w1.setName("yu");
	w1.eat();
	return 0;
}
